bee-message/src/byte_cost.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::output::Output;
+
+use packable::PackableExt;
+
+/// Specifies the current parameters for the byte cost computation.
+#[derive(Clone)]
+pub struct ByteCostConfig {
+    /// Cost in tokens coins per virtual byte.
+    pub byte_cost: f32,
+    /// The weight factor used for key fields in the ouputs.
+    pub weight_for_key: u64,
+    /// The weight factor used for data fields in the ouputs.
+    pub weight_for_data: u64,
+}
+
+impl Default for ByteCostConfig {
+    fn default() -> Self {
+        Self {
+            byte_cost: 1.0,
+            weight_for_key: 10,
+            weight_for_data: 1,
+        }
+    }
+}
+
+/// A trait to facilitate the computation of the byte cost of message outputs, which is central to dust protection.
+pub trait ByteCost {
+    /// Different fields in a type lead to different storage requirements on the tangle.
```

Comment:
Kinda true, but not really what this is about since we can just prune the Tangle
Should be `ledger state`/`UTXO set` or something like that